src/snowflake/snowpark/ml/transformer.py

```diff
@@ -0,0 +1,499 @@
+            else None
+        )
+        self._states_table_name = random_name_for_temp_object(TempObjectType.TABLE)
+
+    @Transformer.input_cols.setter
+    def input_cols(self, value: List[str]) -> None:
+        self._input_cols = value
+        self._states_table_cols = (
+            [f"states_{input_col}" for input_col in self._input_cols]
+            if self._input_cols
+            else None
+        )
+
+    def fit(self, df: DataFrame) -> "StandardScaler":
+        super().fit(df)
+        scaler_fit(self, df, ["mean", "stddev_pop"])
+        return self
+
+    def transform(self, df: DataFrame) -> DataFrame:
+        """
+        The standard score of a sample x is calculated as:
+        z = (x - mean) / stddev
+        """
+        super().transform(df)
+        states_table = df._session.table(self._states_table_name)
+        res_column = df.columns
+        for input_col, states_col, output_col in zip(
+            self.input_cols, self._states_table_cols, self.output_cols
+        ):
+            # if stddev equals to 0, we will set it to 1 to avoid divided by 0 error
+            res_column.append(
+                (
+                    (df[input_col] - col(states_col)["mean"])
+                    / iff(
+                        col(states_col)["stddev_pop"] == 0,
+                        1,
+                        col(states_col)["stddev_pop"],
+                    )
+                ).as_(output_col)
+            )
+
+        df_transform = states_table.join(df).select(res_column)
+        return df_transform
+
+
+class MinMaxScaler(Transformer):
+    def __init__(
+        self,
+        feature_range: Tuple[float, float] = (0, 1),
+        input_cols: Optional[List[str]] = None,
+        output_cols: Optional[List[str]] = None,
+    ) -> None:
+        super().__init__(input_cols, output_cols)
+        if feature_range[0] >= feature_range[1]:
+            raise ValueError(
+                "Maximum  can not be equal or greater than minimum in feature range"
+            )
+        self._states_table_cols = (
+            [f"states_{input_col}" for input_col in self._input_cols]
+            if self._input_cols
+            else None
+        )
+        self.minimum, self.maximum = feature_range
+        self._states_table_name = random_name_for_temp_object(TempObjectType.TABLE)
+
+    @Transformer.input_cols.setter
+    def input_cols(self, value: List[str]) -> None:
+        self._input_cols = value
+        self._states_table_cols = (
+            [f"states_{input_col}" for input_col in self._input_cols]
+            if self._input_cols
+            else None
+        )
+
+    def fit(self, df: DataFrame) -> "MinMaxScaler":
+        super().fit(df)
+
+        scaler_fit(self, df, ["min", "max"])
+        return self
+
+    def transform(self, df: DataFrame) -> DataFrame:
+        """
+        The transformation is given by:
+        X_std = (X - X.min(axis=0)) / (X.max(axis=0) - X.min(axis=0))
```

Comment:
```suggestion
        X_std = (X - min(X)) / (max(X) - min(X))
```
Let's avoid numpy expression since we don't have axis concept.